### ISSUE TITLE
Brocade   BR-VDX6740 making vlan management process crash

### ISSUE THREAD

**Issue by roiKosmic** [NetOpsCommunity/imc-scripts#3]
Hello,
I have integrated your script for ICC backup management to backup my BR-VDX6740 and it works well. Unfortunatly adding the device to iMC make the vlandm component to crash. I have looked at the logs and I think they need specific adapter scripts. Did you write or plan to write them ? or is there any tricks to prevent vlandm to scan vlan on brocade ?

**Comment by LindsayHill** [comment]
Which version of iMC is that? I'll see if I can reproduce it

**Comment by roiKosmic** [comment]
Sorry for the late reply, I am running Imc Plat 7.0 E0102.
Here is the error logs of the vlan management services when the brocade is added (imcvlandm.log)

2017-01-25 00:09:09.023 [INFO (0)] [THREAD(3038144208)] [CQvApplication::start]  call init().
2017-01-25 00:09:09.051 [INFO (0)] [THREAD(3038144208)] [CComponentAdapter::parseVendorAdapter()] Begin to parse ...
2017-01-25 00:09:09.052 [INFO (0)] [THREAD(3038144208)] [CComponentAdapter::parseVendorAdapter()] Finished to parse!
2017-01-25 00:09:09.052 [INFO (0)] [THREAD(3038144208)] [CComponentAdapter::parseVendorAdapter()] Begin to parse ...
2017-01-25 00:09:09.062 [INFO (0)] [THREAD(3038144208)] [CComponentAdapter::parseVendorAdapter()] Finished to parse!
2017-01-25 00:09:09.086 [INFO (0)] [THREAD(3038144208)] [CComponentAdapter::parseVendorAdapter()] Begin to parse ...
2017-01-25 00:09:09.088 [INFO (0)] [THREAD(3038144208)] [CComponentAdapter::parseVendorAdapter()] Finished to parse!
2017-01-25 00:09:09.088 [INFO (0)] [THREAD(3038144208)] [CComponentAdapter::parseVendorAdapter()] Begin to parse ...
2017-01-25 00:09:09.094 [INFO (0)] [THREAD(3038144208)] [CComponentAdapter::parseVendorAdapter()] Finished to parse!
2017-01-25 00:09:09.128 [INFO (0)] [THREAD(3038144208)] [CComponentAdapter::parseVendorAdapter()] Begin to parse ...
2017-01-25 00:09:09.130 [INFO (0)] [THREAD(3038144208)] [CComponentAdapter::parseVendorAdapter()] Finished to parse!
2017-01-25 00:09:09.130 [INFO (0)] [THREAD(3038144208)] [CComponentAdapter::parseVendorAdapter()] Begin to parse ...
2017-01-25 00:09:09.153 [INFO (0)] [THREAD(3038144208)] [CComponentAdapter::parseVendorAdapter()] Finished to parse!
2017-01-25 00:09:09.204 [INFO (0)] [THREAD(3038144208)] [CComponentAdapter::parseVendorAdapter()] Begin to parse ...
2017-01-25 00:09:09.205 [INFO (0)] [THREAD(3038144208)] [CComponentAdapter::parseVendorAdapter()] Finished to parse!
2017-01-25 00:09:09.205 [INFO (0)] [THREAD(3038144208)] [CComponentAdapter::parseVendorAdapter()] Begin to parse ...
2017-01-25 00:09:09.205 [INFO (0)] [THREAD(3038144208)] [CComponentAdapter::parseVendorAdapter()] Finished to parse!
2017-01-25 00:09:09.212 [INFO (0)] [THREAD(3038144208)] [CComponentAdapter::parseVendorAdapter()] Begin to parse ...
2017-01-25 00:09:09.214 [INFO (0)] [THREAD(3038144208)] [CComponentAdapter::parseVendorAdapter()] Finished to parse!
2017-01-25 00:09:09.214 [INFO (0)] [THREAD(3038144208)] [CComponentAdapter::parseVendorAdapter()] Begin to parse ...
2017-01-25 00:09:09.224 [INFO (0)] [THREAD(3038144208)] [CComponentAdapter::parseVendorAdapter()] Finished to parse!
2017-01-25 00:09:09.255 [INFO (0)] [THREAD(3038144208)] [CComponentAdapter::parseVendorAdapter()] Begin to parse ...
2017-01-25 00:09:09.256 [INFO (0)] [THREAD(3038144208)] [CComponentAdapter::parseVendorAdapter()] Finished to parse!
2017-01-25 00:09:09.256 [INFO (0)] [THREAD(3038144208)] [CComponentAdapter::parseVendorAdapter()] Begin to parse ...
2017-01-25 00:09:09.257 [INFO (0)] [THREAD(3038144208)] [CComponentAdapter::parseVendorAdapter()] Finished to parse!
2017-01-25 00:09:09.269 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: max-length
2017-01-25 00:09:09.269 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: illegal-chars
2017-01-25 00:09:09.269 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: first-valid-char
2017-01-25 00:09:09.269 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: unique
2017-01-25 00:09:09.271 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: max-length
2017-01-25 00:09:09.271 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: illegal-chars
2017-01-25 00:09:09.271 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: first-valid-char
2017-01-25 00:09:09.271 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: unique
2017-01-25 00:09:09.273 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: max-length
2017-01-25 00:09:09.273 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: illegal-chars
2017-01-25 00:09:09.273 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: first-valid-char
2017-01-25 00:09:09.273 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: unique
2017-01-25 00:09:09.274 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: max-length
2017-01-25 00:09:09.275 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: illegal-chars
2017-01-25 00:09:09.275 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: first-valid-char
2017-01-25 00:09:09.275 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: unique
2017-01-25 00:09:09.276 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: max-length
2017-01-25 00:09:09.276 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: illegal-chars
2017-01-25 00:09:09.276 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: first-valid-char
2017-01-25 00:09:09.276 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: unique
2017-01-25 00:09:09.285 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: max-length
2017-01-25 00:09:09.285 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: illegal-chars
2017-01-25 00:09:09.285 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: first-valid-char
2017-01-25 00:09:09.285 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: unique
2017-01-25 00:09:09.307 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: max-length
2017-01-25 00:09:09.307 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: illegal-chars
2017-01-25 00:09:09.307 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: first-valid-char
2017-01-25 00:09:09.307 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: unique
2017-01-25 00:09:09.315 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: max-length
2017-01-25 00:09:09.315 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: illegal-chars
2017-01-25 00:09:09.315 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: first-valid-char
2017-01-25 00:09:09.315 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: unique
2017-01-25 00:09:09.317 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: max-length
2017-01-25 00:09:09.317 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: illegal-chars
2017-01-25 00:09:09.317 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: first-valid-char
2017-01-25 00:09:09.317 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: unique
2017-01-25 00:09:09.319 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: max-length
2017-01-25 00:09:09.319 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: illegal-chars
2017-01-25 00:09:09.319 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: first-valid-char
2017-01-25 00:09:09.319 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: unique
2017-01-25 00:09:09.321 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: max-length
2017-01-25 00:09:09.321 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: illegal-chars
2017-01-25 00:09:09.321 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: first-valid-char
2017-01-25 00:09:09.321 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: unique
2017-01-25 00:09:09.331 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: max-length
2017-01-25 00:09:09.331 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: illegal-chars
2017-01-25 00:09:09.331 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: first-valid-char
2017-01-25 00:09:09.331 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: unique
2017-01-25 00:09:09.434 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: max-length
2017-01-25 00:09:09.434 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: illegal-chars
2017-01-25 00:09:09.434 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: first-valid-char
2017-01-25 00:09:09.434 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: unique
2017-01-25 00:09:09.443 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: max-length
2017-01-25 00:09:09.443 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: illegal-chars
2017-01-25 00:09:09.444 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: first-valid-char
2017-01-25 00:09:09.444 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: unique
2017-01-25 00:09:09.445 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: max-length
2017-01-25 00:09:09.445 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: illegal-chars
2017-01-25 00:09:09.445 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: first-valid-char
2017-01-25 00:09:09.445 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: unique
2017-01-25 00:09:09.447 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: max-length
2017-01-25 00:09:09.447 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: illegal-chars
2017-01-25 00:09:09.447 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: first-valid-char
2017-01-25 00:09:09.447 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: unique
2017-01-25 00:09:09.454 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: max-length
2017-01-25 00:09:09.454 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: illegal-chars
2017-01-25 00:09:09.454 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: first-valid-char
2017-01-25 00:09:09.454 [WARNING (0)] [THREAD(3038144208)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: unique

**Comment by LindsayHill** [comment]
7.0 is now > 3 yrs old. I know there have been a lot of fixes since then, including some that were related to non-alpha characters in VLAN names. 

I'll spin up a 7.3 system and see if it reports any issues with a VDX. I don't recall seeing any issues when trying it with 7.1/7.2, but it could be related to VLAN names, and my lab would have had few names defined. 

I suggest you start planning an upgrade anyway. Lots of new features, lots of bug fixes and security fixes in 2013 when 7.0 was released.

**Comment by roiKosmic** [comment]
Thanks for your reply. I am going to analyse how to upgrade and best way to handle this migration.
Vincent

**Comment by LindsayHill** [comment]
If you install a clean IMC 7.3 system, you see similar warnings to the above - looks like it happens for the ZTE adapter:
```
2017-01-31 18:47:36.161 [INFO (0)] [THREAD(4111677184)] [CVLanStyleMgr::initCmdVlanFuncs] /opt/iMC/server/bin/../../server/conf/adapters/vlandm/ZTE/ZXR10/vlan_func_def.xml not exist.
2017-01-31 18:47:36.161 [WARNING (0)] [THREAD(4111677184)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: max-length
2017-01-31 18:47:36.161 [WARNING (0)] [THREAD(4111677184)] [CGeneralVlanXMLParser::parseVlanNameProperty] receive unknown node and discard, node name: illegal-chars```

This does not cause imcvlandm to crash, so I think we can ignore those messages. 

I've just done some tests with 7.3 E0504, and around 20 VDX switches. They seem to import OK, and I'm not seeing any crashes with imcvlandm.

As I understand it, the VLAN adapter files are for if you want to do things like actively manage VLANs on those devices. Shouldn't be a problem if you're not changing any VLAN configs.

This here might be related: http://h20564.www2.hpe.com/hpsc/doc/public/display?docId=mmr_kc-0122724

**Comment by roiKosmic** [comment]
Hello, Just for info I have installed last released of 7.3. But I had some issues with ACL module (retrieving ALC from existing device failed) HP told me the 7.3 was still buggy and advice to still use 7.2
Everything is working with 7.2 thanks for your help.
Vincent

**Comment by LindsayHill** [comment]
Good to hear 7.2 is working (and interesting to know about issues with 7.3)
